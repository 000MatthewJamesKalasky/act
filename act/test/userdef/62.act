deftype testme (int<4> x; chan(int) y)
{

}


function foo (testme t) : bool
{
   chp {
     self := true;
     t.y!1
   }
}


defproc test()
{
   testme x;
   bool y;
   chp {
     y := foo(x)
   }
}
