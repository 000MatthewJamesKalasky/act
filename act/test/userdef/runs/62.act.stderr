ERROR: File `62.act', line 7, col 24
	Parameter `t': port parameter for a function cannot be a structure with channels
